hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java

```diff
@@ -96,9 +96,6 @@ protected void destroy() throws Exception {
     }
 
     protected final boolean runScheduledTasks() {
-        final PriorityQueue<ScheduledTask> scheduledTaskQueue = this.scheduledTaskQueue;
```

Comment:
These local fields need to be restored. I saw in my benchmarks that this can have an impact on performance.